consumer.go

```diff
@@ -392,6 +394,31 @@ func (child *partitionConsumer) HighWaterMarkOffset() int64 {
 	return atomic.LoadInt64(&child.highWaterMarkOffset)
 }
 
+func (child *partitionConsumer) responseFeeder() {
+	for response := range child.feeder {
+		switch err := child.handleResponse(response); err {
+		case nil:
+			child.broker.acks <- nil
+		case ErrOffsetOutOfRange:
+			// there's no point in retrying this it will just fail the same way again
+			// so shut it down and force the user to choose what to do
+			child.AsyncClose()
+			fallthrough
+		default:
+			child.sendError(err)
+			fallthrough
+		case ErrUnknownTopicOrPartition, ErrNotLeaderForPartition, ErrLeaderNotAvailable:
+			// these three are not fatal errors, but do require redispatching
+			Logger.Printf("consumer/%s/%d abandoned broker %d because %s\n", child.topic, child.partition, child.broker.broker.ID(), err)
+			child.broker.acks <- child
+			child.trigger <- none{}
+		}
```

Comment:
The alternative is to duplicate these three lines twice (and the `sendError` call one additional time). Using `fallthrough` explicitly, (as opposed to the stupid implicit C version) seems much more elegant to me.

I don't want to derail the PR (as you mentioned, this code is just moved, not changed), but you've piqued my curiousity - what bothers you about this pattern?